src/main/resources/com/atlassian/performance/tools/report/distribution/distribution-comparison-template.html

```diff
@@ -19,7 +19,32 @@
 <canvas id="histogram"></canvas>
 <canvas id="quantile-chart"></canvas>
 <script>
+    function groupBy(arr, fn) {
+        return arr.reduce(function(acc, e) {
+            var key = fn(e);
+
+            if (!acc[key])
+                acc[key] = [e];
+            else
+                acc[key].push(e);
+
+            return acc;
+        }, {});
+    }
+
+    function flattenGroups(groups) {
+        return Object.values(groups).reduce(function (acc, e) {
+            return acc.concat(e);
+        }, []);
+    }
+
     function jptChartTemplate(data) {
+        var datasetGroups = groupBy(data.datasets, function (dataset) {
+            return dataset.label.replace(/^(baseline|experiment):\s*/, '');
```

Comment:
Quoting @wyrzyk :
> The “baseline/experiment” labels are specific for regression tests. For other tests it can be a different label.
> See how it’s created in `DistributionComparison` : `label = "${result.cohort}: $actionType"`

In order to handle dynamic `result.cohort`, we can reshape the data from the Kotlin side in the `DistributionComparison` class.

Reply:
Did this in the manner of minimal change, hence simply added a `cohort` field to the `ChartLine` and wrapped the corresponding JSON field with a check to prevent it from being shown when it is blank (essentially, only showing it for `distribution-comparison`).

Reply:
I like how you control the impact of the change, well done 👍 
JS code is now simpler and therefore there's even less reason to port the logic to Kotlin.
The important part is that it works and we have tests to prove it. Let's ship it 🚀